File: 98-Validate-Binary-Search-Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    inlist = []

    def inorder(root):
        if root is None:
            return
        Solution.inorder(root.left)
        Solution.inlist.append(root.val)
        Solution.inorder(root.right)

    def isValidBST(self, root: Optional[TreeNode]) -> bool:
        Solution.inorder(root)
        if len(Solution.inlist) <= 1:
            return True
        for i, _ in enumerate(Solution.inlist[:-1]):
            if Solution.inlist[i] >= Solution.inlist[i + 1]:
                return False
        return True

File: 98-Validate-Binary-Search-Tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {boolean}
 */
const isValidBST = function (root) {
  let inlist = []
  function inorder(root) {
    if (!root) return
    inorder(root.left)
    inlist.push(root.val)
    inorder(root.right)
  }
  inorder(root)
  for (let i = 0; i < inlist.length - 1; ++i) {
    if (inlist[i] >= inlist[i + 1]) return false
  }
  return true
}
